rfcs/text/0014_scheduled_reports_new_plugin.md

```diff
@@ -0,0 +1,164 @@
+
+```ts
+interface IRegisterReportOptions {
+  // Unique ID to identify the job
+  id: string;
+
+  // List of arbitrary tags to perform filtering for in reporting's UI
+  tags: string[];
+
+  // The desired type of asset once complete
+  type: 'pdf' | 'png' | 'csv';
+
+  // TBD: The label of the button on the page ("Export Page")
+  label: string;
+
+  // Enumerable of "who" can see/download the report.
+  visibility: 'user' | 'public';
+
+  // Available intervals of exporting, 'immediate' equating to what we currently have now
+  intervals: Array<'immediate' | 'monthly' | 'weekly' | 'daily'>;
+
+  // A list of connectors to use for exporting
+  connectors: Array<'email' | 'slack' | 'iqcu'>;
+
+  // A before hook function that gets the full context of the server-side request. Returns a true/false indicating whether or not to proceed.
+  before: (context: RequestHandlerContext, req: KibanaRequest, res: KibanaResponse) => Promise<boolean>;
+
+  // A post-request hook that allows altering or viewing the produced asset, as a buffer, of the job.
+  after: (result: Buffer) => void;
+}
+
+interface IReportingClient {
+  // Open triggers the generation report UI to open, and if it's configurable to be scheduled, that UI as well.
+  // The argument here is _where_ the asset is located (either a fully-qualified URL or saved-object ID).
+  open: (urlOrObjectID: string) => void;
+
+  // Simply close the report generation panel.
+  close: () => void;
+}
+```
+
+## 4. Render your page properly
+
+In cases where a report is a rasterized asset (PDF or PNG), your page should take care of when and how this happens. For instance, if your page is located at:
+
+```
+https://www.kibana-is-cool.com/my-dope-kibana-page-a-rama
+```
+
+Then, setting a URL-based argument might make sense in your hook:
+
+```
+https://www.kibana-is-cool.com/my-dope-kibana-page-a-rama?pdf=true
+```
+
+Having something to distinguish whether a page is being exported, or simply rendered as usual, is critical and allows page owners to make conditional decisions based upon that predicate. For instance, app owners can (and should) disable "first-time" wizards, telemetry, and anything else expecting user interactions (modals/prompts/buttons/etc). The browser that reporting uses will appear as a "first-time" user with no cache or cookies set.
+
+## 5. ... Wait
+
+After a report has been submitted, either via a user-initiated action or a programmatic one (not covered in this RFC), reporting will function like it generally has. Changes here are purely internal to the team, and don't require commentary since the implementation isn't of concern to consumers. Once complete, a toast-notification is trigger with a link to download the report (in the case of immediate intervals, or simple no-connector) and a link to the Reporting page. For scheduled reports, a simple success notification is generated at the start to give notice to the user that the report will begin at the requested interval.
```

Comment:
Currently `ScreenCapturePanelContent` handles showing the toast notifications, so this ease-of-use for developers already exists for manually generated reports.